src/test/java/itm/oss/splitter/AppTest.java

```diff
@@ -0,0 +1,99 @@
+package itm.oss.splitter;
+
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.math.BigDecimal;
+
+import org.junit.jupiter.api.DisplayName;
+import java.io.ByteArrayInputStream;
+
+public class AppTest {
+
+    // JUnit 5 기본 구조
+    @Test
+    void testApp_Placeholder() {
+        assertTrue(true); 
+    }
+
+    private final InputStream originalIn = System.in;
+    private final PrintStream originalOut = System.out;
+
+    private App createAppWithSimulatedInput(String input) {
+        System.setIn(new ByteArrayInputStream(input.getBytes()));
+        return new App();
+    }
+
+    @Test
+    @DisplayName("Edge Case: readRequiredLine rejects empty input")
+    void testReadRequiredLine_EmptyInput_ShowsError(){
+        ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outContent));System.setOut(new PrintStream(outContent));
+
+        try{ 
+            App app = createAppWithSimulatedInput("\n\nGithero\n");
+            String result = app.readRequiredLine("Enter name: ");
+            String output = outContent.toString();
+
+            assertEquals("Githero", result);
+            assertTrue(output.contains("This field is required and cannot be empty."));
+        } finally { 
+        System.setIn(originalIn);
+        System.setOut(originalOut);
+        }
+    }
+
+    @Test
+    @DisplayName("Edge Case: readBigDecimal rejects invalid amounts")
+    void testReadBigDecimal_InvalidAmount_ShowsError() {
+        ByteArrayOutputStream outContent = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(outContent));
+
+        try{
+            App app = createAppWithSimulatedInput("abc\n0\n-100\n50.5\n");
+
+            BigDecimal result = app.readBigDecimal("Enter amount: ");
+            String output = outContent.toString();
+
+            assertEquals(new BigDecimal("50.5"), result);
```

Comment:
Good job verifying that the valid numeric input is correctly parsed. 
However, when comparing BigDecimal values, assertEquals() can sometimes fail if the scales differ(ex. 50.5 vs 50.50).
Consider using different way instead, which ignores scale differences.

```suggestion
assertTrue(result.compareTo(new BigDecimal("50.5")) == 0)
```

Reply:
Thanks for leaving critical suggestion. We missed the potential issue of scale differences between two. I've applied the suggestion to replace `assertEquals(new BigDecimal("50.5"), result);` with `assertTrue(result.compareTo(new BigDecimal("50.5")) == 0)` as you suggested. I pushed the change. Could you please check my commit again for approval?

Reply:
LGTM. Now it looks more comfortable to me. Thanks!